src/wellKnownDidConfiguration/wellKnownDidConfiguration.test.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  KeyringPair,
+  DidUri,
+  DidDocument,
+  ICredentialPresentation,
+} from '@kiltprotocol/sdk-js'
+import { ApiPromise } from '@polkadot/api'
+import { mnemonicGenerate, cryptoWaitReady } from '@polkadot/util-crypto'
+import { VerifiableDomainLinkagePresentation } from '../types/types'
+import { BN } from '@polkadot/util'
+import { Keyring } from '@kiltprotocol/utils'
+import {
+  createCredential,
+  getDomainLinkagePresentation,
+  verifyDidConfigPresentation,
+} from './wellKnownDidConfiguration'
+import {
+  fundAccount,
+  generateDid,
+  buildConnection,
+  keypairs,
+  createCtype,
+  assertionSigner,
+} from '../tests/utils'
+
+let api: ApiPromise
+
+describe('Well Known Did Configuration integration test', () => {
+  let mnemonic: string
+  let account: KeyringPair
+  const origin = 'http://localhost:3000'
+  let didDocument: DidDocument
+  let didUri: DidUri
+  let keypair: any
+  let domainLinkageCredential: VerifiableDomainLinkagePresentation
+  let credential: ICredentialPresentation
+
+  beforeAll(async () => {
+    api = await buildConnection('ws://127.0.0.1:9944')
+  })
+
+  beforeAll(async () => {
+    await cryptoWaitReady()
+    mnemonic = mnemonicGenerate()
+    account = new Keyring({ type: 'ed25519' }).addFromMnemonic(mnemonic)
+    await fundAccount(account.address, new BN('1000000000000000000'), api)
+    keypair = await keypairs(account, mnemonic)
+
+    didDocument = await generateDid(account, mnemonic)
+
+    didUri = didDocument.uri
+    await createCtype(didUri, account, mnemonic, api)
+  }, 30_000)
+
+  it('generate a well known did configuration credential', async () => {
+    expect(
+      (credential = await createCredential(
+        await assertionSigner({ assertion: keypair.assertion, didDocument }),
+        origin,
+        didUri
+      ))
+    ).toBeTruthy()
+  }, 30_000)
+
+  it('fails to generate a well known did configuration credential due to bad origin', async () => {
```

Comment:
pls rename the test - from the code I understand we check that this fails if the origin parameter is not a URL, but that's impossible to understand from the test alone

Reply:
Updated.